### ISSUE TITLE
Playback not working on Kodi 17/17.1 Libreelec 8/8.1

### ISSUE THREAD

**Issue by Dazr** [Yoshiofthewire/CrunchyXBMC#61]
I am having problems with the Crunchyroll addon (with both versions - the version in the Kodi repository and the latested version from GitHub) with Libreelec 8.0.1 (Kodi 17.1) on RPi 3. (The same problem happens on both Kodi 17 and 17.1).

The addon runs, but after about 30 seconds of playing video, it begins to buffer, then plays, then loses all audio while playing back a static image once every 2 seconds (as if fast forwarding - but I'm not fast forwarding). The behaviour is almost similar to not having enough bandwidth to play smoothly, but there is plenty of bandwidth. If I revert back to an earlier version of Openelec and Kodi (15), it's absolutely smooth playback with no problems at all. So the problem has happened only since upgrading to Kodi 17.

Note that the Video quality setting for all of this is 480p. I even tried 360p, it's slightly better, but still the same problem occurs.

(Note that I am on a good quality ADSL2+ connection (not Fibre - maybe this issue doesn't come to light with a 30Mbps+ Fibre connection?), and haven't had any issues with Crunchyroll on ADSL2+ in the past. And if I go back to a previous version of Kodi before 17, everything works great. It's just since Kodi 17 that this prolbem now happens). I get 1.3 MB (Megabytes not megabits) down which is very good for ADSL2+ and can stream higher resolution video from other sources no problem).

Thanks for your help with this!

**Comment by davidkna** [comment]
Essentially to facilitate playback the addon only digs up the playback url and tells Kodi to play it. Thus I think this might be an issue with Kodi. Do you have this issue with any other addons?

**Comment by Dazr** [comment]
All of my other video streaming addons are working fine without any problem.  I see with Kodi 17 there were a lot of changes - from the changelog:

"1.4 Platform Independent Playback

    Major refactoring of Kodi's built-in video player core (which also been renamed from "DVDPlayer" to "VideoPlayer")[54][55][56]
    Added support for "InputStream" binary addon extension point that can serve as input stream for VideoPlayer[57][58]
    Added support for "InputStream" (input stream) addons to system addons[59]
    Added "InputStream" (input stream) client addons for common network streaming protocols and popular services:
        InputStream Addon (default reference InputStream Addon by Team Kodi
        Test InputStream Addon (Test InputStream Addon for Kodi's new inputstream interface
        Adaptive InputStream Addon (InputStream Client addon for several manifest types)
        RTMP InputStream Addon (InputStream Client for RTMP streams using librtmp)
        MPD (MPEG-DASH) InputStream Addon
        SmoothStream InputStream Addon for MS-SSTR (SmoothStreamingMedia) Smooth Streaming Protocol
        NXMSL InputStream Addon for Netflix style manifests over MSL"

So I guess maybe this has something to do with the problem.

Basically, what happens when I try to play the video, the (for want of a better word) ' please wait icon' appears and starts spinning for about 14 seconds.  Then it completely stops spinning and stays static for about 5 seconds.  Then it kind of flashes (disappears then re-appears quickly) and starts spinning again for about 5 seconds and plays the video.  Almost as if the first call to 'play' fails, and then it falls back to a second call to 'play' (just an assumption).

**Comment by MrKrabat** [comment]
I am using OSMC and with Kodi 17/17.1 there were no issues with playback on RPi 3.
May this problem is related to LibreELEC?

In the past i had problems with a bad WLAN performance of the RPi 3, can you try to use a cable connection?

**Comment by 0nix** [comment]
+1 on this issue. I had to switch back to Kodi 15.2 Isengard to get the Add-On to work at normal performance. The WLAN may have been an issue. I will conduct a test with Kodi Krypton and a wired connection.
